collect_app/src/main/java/org/odk/collect/android/external/AndroidShortcutsActivity.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2011 University of Washington
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.odk.collect.android.external
+
+import android.content.DialogInterface
+import android.content.Intent
+import android.os.Bundle
+import android.os.Parcelable
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import org.odk.collect.android.R
+import org.odk.collect.android.analytics.AnalyticsEvents
+import org.odk.collect.android.analytics.AnalyticsUtils
+import org.odk.collect.android.formlists.blankformlist.BlankFormListItem
+import org.odk.collect.android.formlists.blankformlist.BlankFormListViewModel
+import org.odk.collect.android.injection.DaggerUtils
+import org.odk.collect.settings.SettingsProvider
+import javax.inject.Inject
+
+/**
+ * Allows the user to create desktop shortcuts to any form currently available to Collect
+ */
+class AndroidShortcutsActivity : AppCompatActivity() {
+    @Inject
+    lateinit var viewModelFactory: BlankFormListViewModel.Factory
+
+    @Inject
+    lateinit var settingsProvider: SettingsProvider
+
+    private val viewModel: BlankFormListViewModel by viewModels { viewModelFactory }
+
+    public override fun onCreate(bundle: Bundle?) {
+        super.onCreate(bundle)
+        DaggerUtils.getComponent(this).inject(this)
+
+        viewModel.formsToDisplay.observe(this) { forms -> showFormListDialog(forms) }
```

Comment:
I think this could crash if the forms load after this observation occurs as value of `formsToDisplay` could be `null`. The real problem there is just that `formsToDisplay` should be `LiveData<List<BlankFormListItem?>>` instead of `LiveData<List<BlankFormListItem>>` I think.

Reply:
`formsToDisplay` has no initial value so observers won't be notified until data is loaded.

Reply:
Could we still change the type and correct the observer accordingly? I've realized that unless we're using `NonNullLiveData` we should always be marking the `LiveData` type as `?` as there's no (compiler level) protection against a `null` creeping in.

Reply:
So you want to have type like `MutableLiveData<List<BlankFormListItem>?>` and always check if the list is not null?

Reply:
Yeah, I think that's best! It's annoying that `LiveData` doesn't support optional typing properly. It looks like some people are advocating a switch to [`StateFlow`](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-state-flow/) in Kotlin for that reason although I haven't done a lot of reading on it.

Reply:
Ok so I added it in https://github.com/getodk/collect/pull/5077/commits/94243c02d87e4781e3c2aa219d9ed8bfaac3afd4